Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -188,8 +188,9 @@ final class ConfigGenerator: ConfigGenerating {
         }
         settings["SDKROOT"] = .string(target.platform.xcodeSdkRoot)
         settings["SUPPORTED_PLATFORMS"] = .string(target.platform.xcodeSupportedPlatforms)
-        // TODO: We should show a warning here
+
         if settings["SWIFT_VERSION"] == nil {
+            logger.log(level: .warning, "Setting the build setting SWIFT_VERSION to \(swiftVersion) for target '\(target.name)'. We recommend setting it on the manifest.")
```

Comment:
The warning will always be displayed even if `SWIFT_VERSION ` is set, as `settings` is a local dictionary in this method - it may be more suited for inclusion in the `SettingsLinter`.

Perhaps a note in the change log / release notes would suffice instead of a warning.

e.g. this feels a bit too noisy :( 

```
$ swift run tuist generate
Generating workspace App.xcworkspace
Generating project App
Setting the build setting SWIFT_VERSION to 5.1.2 for target 'App'. We recommend setting it on the manifest.
Setting the build setting SWIFT_VERSION to 5.1.2 for target 'App'. We recommend setting it on the manifest.
Setting the build setting SWIFT_VERSION to 5.1.2 for target 'AppTests'. We recommend setting it on the manifest.
Setting the build setting SWIFT_VERSION to 5.1.2 for target 'AppTests'. We recommend setting it on the manifest.
Setting the build setting SWIFT_VERSION to 5.1.2 for target 'AppUITests'. We recommend setting it on the manifest.
Setting the build setting SWIFT_VERSION to 5.1.2 for target 'AppUITests'. We recommend setting it on the manifest.
Project generated.
Total time taken: 1.362s
```

Reply:
Ouch, good catch. The warning is tricky because it might be defined in a .xcconfig file and we'd need to parse them to be 100% accurate with the warning. I just removed it for now.